--- a/setup.py
+++ b/setup.py
@@ -19,7 +19,6 @@ License along with this library.
 from __future__ import print_function
 from __future__ import absolute_import
 from __future__ import division
-from __future__ import unicode_literals
 from distutils.core import setup
 
 setup(
@@ -30,5 +29,7 @@ setup(
     author_email = ["enrico@enricozini.org"],
     url = "https://github.com/spanezz/django-housekeeping",
     license = "https://www.gnu.org/licenses/lgpl.html",
-    packages = ["django_housekeeping"],
+    packages = ["django_housekeeping",
+                "django_housekeeping.management",
+                "django_housekeeping.management.commands"]
 )
